pkg/streamer/reader.go

```diff
@@ -71,12 +74,16 @@ func NewBinlogReader(cfg *BinlogReaderConfig) *BinlogReader {
 	if cfg.Timezone != nil {
 		parser.SetTimestampStringLocation(cfg.Timezone)
 	}
+
+	newtctx := tctx.WithContext(ctx)
+	newtctx = newtctx.WithLogger(tctx.L().WithFields(zap.String("component", "binlog reader")))
```

Comment:
Is it better to add another function, such as `NewContext(ctx context.Context, logger log.Logger) *tcontext.Context` to return a `*tcontext.Context` directly. So we don't need to create two `*tcontext.Context` instances like this.

Reply:
good suggestion